File: gameoflife/java/src/test/java/gameoflife/CellTest.java
package gameoflife;

import org.junit.jupiter.api.Test;

import java.util.Set;

import static org.junit.jupiter.api.Assertions.assertAll;
import static org.junit.jupiter.api.Assertions.assertTrue;


class CellTest {

    @Test
    void neighbors() {

        Cell sut = new Cell(2, 3);
        Set<Cell> neighbors = sut.neighbors();

        //J-
        assertAll(
                () -> assertTrue(neighbors.contains(new Cell(1, 2))),
                () -> assertTrue(neighbors.contains(new Cell(2, 2))),
                () -> assertTrue(neighbors.contains(new Cell(3, 2))),
                () -> assertTrue(neighbors.contains(new Cell(1, 3))),
                () -> assertTrue(neighbors.contains(new Cell(3, 3))),
                () -> assertTrue(neighbors.contains(new Cell(1, 4))),
                () -> assertTrue(neighbors.contains(new Cell(2, 4))),
                () -> assertTrue(neighbors.contains(new Cell(3, 4))),
                () -> assertTrue(neighbors.size() == 8));
        //J+
    }
}

File: gameoflife/kotlin/src/test/kotlin/gameoflife/CellTest.kt
package gameoflife

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class CellTest {

    @Test
    fun neighbors() {
        val sut = Cell(2, 3)

        val expected = setOf(
                Cell(1, 2), Cell(2, 2), Cell(3, 2),
                Cell(1, 3), Cell(3, 3),
                Cell(1, 4), Cell(2, 4), Cell(3, 4))

        assertEquals(expected, sut.neighbors)
    }
}